src/tracer.cc

```diff
@@ -145,6 +163,16 @@ TraceManager::UpdateTraceSettingInternal(
                                    : (((current_setting != nullptr) &&
                                        current_setting->filepath_specified_) ||
                                       (new_setting.filepath_ != nullptr)));
+  const bool mode_specified =
+      (new_setting.clear_mode_ ? false
+                               : (((current_setting != nullptr) &&
+                                   current_setting->mode_specified_) ||
+                                  (new_setting.mode_ != nullptr)));
+  const bool config_map_specified =
+      (new_setting.config_map_ ? false
+                               : (((current_setting != nullptr) &&
+                                   current_setting->config_map_specified_) ||
+                                  (new_setting.config_map_ != nullptr)));
```

Comment:
Is there some logic off here?

I imagine this would have some symmetry like either:
```
const bool mode_specified = (new_setting.clear_mode_ ...)  // clear var
const bool config_map_specified = (new_setting.clear_config_map_ ...)  // clear var
```
or
```
const bool mode_specified = (new_setting.mode_ ...)  // actual var
const bool config_map_specified = (new_setting.config_map_ ...)  // actual var
```

but currently it is
```
const bool mode_specified = (new_setting.clear_mode_ ...)  // clear var
const bool config_map_specified = (new_setting.config_map_ ...)  // actual var
```

Side note, I don't think the new `clear_` vars are being used as mentioned in my other comment: https://github.com/triton-inference-server/server/pull/5637/files#r1167337834

So if they're not used in this PR, it may simplify the amount of code to review if you remove those parts

Reply:
thanks for catching this.